tools/gcs-bucket-mover/gcs_bucket_mover/configuration.py

```diff
@@ -0,0 +1,151 @@
+        self._target_service_account_email = self.__get_service_account_email(
+            self._target_storage_client)
+
+        # Load the values passed in on the command line
+        self._bucket_name = conf.bucket_name
+        self._temp_bucket_name = self.__get_temp_bucket_name(
+            self._bucket_name, conf.tempBucketName)
+        self._source_project = conf.source_project
+        self._target_project = conf.target_project
+
+    @property
+    def source_project_credentials(self):
+        """Get the credentials object for the source project"""
+        return self._source_project_credentials
+
+    @property
+    def target_project_credentials(self):
+        """Get the credentials object for the target project"""
+        return self._target_project_credentials
+
+    @property
+    def source_storage_client(self):
+        """Get the storage client object for the source project"""
+        return self._source_storage_client
+
+    @property
+    def target_storage_client(self):
+        """Get the storage client object for the target project"""
+        return self._target_storage_client
+
+    @property
+    def target_service_account_email(self):
+        """Get the email account of the target project's service account"""
+        return self._target_service_account_email
+
+    @property
+    def bucket_name(self):
+        """Get the name of the bucket to be moved"""
+        return self._bucket_name
+
+    @property
+    def temp_bucket_name(self):
+        """Get the temporary bucket name to use in the target project"""
+        return self._temp_bucket_name
+
+    @property
+    def source_project(self):
+        """Get the name of the source project"""
+        return self._source_project
+
+    @property
+    def target_project(self):
+        """Get the name of the target project"""
+        return self._target_project
+
+    def __get_service_account_email(self, storage_client):  #Google guidelines differ from pylint. pylint: disable=no-self-use
+        # Copied from an approved PR to
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/blob/master/storage/google/cloud/storage/client.py pylint: disable=line-too-long
+        # Once it is released, this can be replaced with the library call.
+        """Get the email address of the project's GCS service account
+
+        Args:
+            storage_client: The storage client object used to access GCS
+
+        Returns:
+            The service account email address
+        """
+
+        project = storage_client.project
+        path = '/projects/%s/serviceAccount' % (project,)
+        api_response = storage_client._base_connection.api_request(  #pylint: disable=protected-access
+            method='GET', path=path)
+        return api_response['email_address']
+
+    def __get_temp_bucket_name(self, bucket_name, temp_bucket_name):  #Google guidelines differ from pylint. pylint: disable=no-self-use
```

Comment:
I don't know what "Google guidelines differ from pylint" means... but this should be a local function, not a method.

Reply:
pylint didn't want it to be a method and so shouldn't need self, but yapf wouldn't accept it is a function. I just inlined the two lines of code and removed the method.